plugins/carla/CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+cmake_minimum_required(VERSION 3.16...3.25)
+
+legacy_check()
+
+option(ENABLE_CARLA "Enable building OBS with carla plugin host" ON)
+if(NOT ENABLE_CARLA)
+  target_disable(carla)
+  return()
+endif()
+
+# Find Qt
+find_qt(COMPONENTS Core Widgets)
+
+# Use pkg-config to find optional deps
+find_package(PkgConfig)
+
+# Find pthread via cmake
+if(OS_WINDOWS)
+  set(carla_pthread_libs OBS::w32-pthreads)
```

Comment:
You link against `OBS::libobs` which carries a transitive dependency on `w32-pthreads`, so you don't need to explicitly link it yourself.

Reply:
on carla-bridge plugin sure, but the discovery and bridge tools make use of pthreads and do not link against libobs. so the variable is still needed for those.

Reply:
Got it - you can just use `$<$<PLATFORM_ID:Windows>OBS::w32-pthreads>` with `target_link_libraries` to get it as a conditional linking requirement.

You also need to do the following close to the top of `CMakeLists.txt` (because you cannot assume that the target exists by itself):

```
if(NOT TARGET OBS::w32-pthreads)
  add_subdirectory("${CMAKE_SOURCE_DIR}/deps/w32-pthreads" "${CMAKE_BINARY_DIR}/deps/w32-pthreads")
endif()
```

This is still a bit brittle `CMAKE_SOURCE_DIR` is not assured to be the checkout's root directory, but it's good enough for now.

Reply:
carla side now handles the building of the tools, so it uses its own local static pthreads.
this section no longer applies.